src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,111 @@
+package duke.storage;
+
+import duke.data.exception.DukeException;
+import duke.data.task.*;
+
+import java.io.FileWriter;
+import java.io.File;
+//import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class that deals with loading tasks from the file and saving tasks in the file.
+ *
+ * @author Won Ye Ji
+ */
+public class Storage {
+    private final String filePath;
+    private final File file;
+
+    /**
+     * Constructor for Storage class.
+     *
+     * @param filePath The filepath of the stored file.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        file = new File(filePath);
+        try {
+            file.getParentFile().mkdirs();
+            file.createNewFile();
+        } catch (IOException e) {
+            System.err.println("IOException: Unable to create file.");
+        }
+    }
+
+    /**
+     * Loads the tasks from the file.
+     *
+     * @return An arraylist of the tasks.
+     * @throws DukeException if the tasks cannot be loaded.
+     */
+    public ArrayList<Task> loadTasks() throws DukeException {
+        try {
+            Scanner sc = new Scanner(file);
+            ArrayList<Task> list = new ArrayList<>();
+            while (sc.hasNext()) {
+                Task task = taskFormatter(sc.nextLine());
+                list.add(task);
+            }
+            return list;
+        } catch (IOException e) {
+            System.err.println("Error loading file.");
+            return new ArrayList<Task>();
+        }
+    }
+
+    /**
+     * Updates the file according to the tasklist.
+     *
+     * @param tasks String representatio of all tasks.
+     * @throws DukeException if unexpected error is encountered.
+     */
+    public void updateFile(String tasks) throws DukeException {
+        try {
+            FileWriter fw = new FileWriter(filePath);
+            fw.write(tasks);
+            fw.close();
+        } catch (IOException e) {
+            throw new DukeException("Unexpected error.");
+        }
+    }
+
+    /**
+     * Returns a task that corresponds with the string description of the task.
+     *
+     * @param task String representation of the task to be returned.
+     * @return A task that corresponds with the string description of the task.
+     * @throws DukeException if task cannot be identified.
+     */
+    public Task taskFormatter(String task) throws DukeException {
+        String type = task.substring(1, 2);
+        String doneStatus = task.substring(4, 5);
+        Task t;
```

Comment:
Instead of naming it `t`, perhaps a more appropriate name will be `newTask`?
```suggestion
        Task newTask;
```